nestkernel/synaptic_element.h

```diff
@@ -31,6 +31,48 @@
 #ifndef SYNAPTIC_ELEMENT_H
 #define SYNAPTIC_ELEMENT_H
 
+/* BeginDocumentation
+  Name: synaptic_element - Contact point element for the dynamic creation
+   and deletion of synapses.
+
+  Description:
+   This class represents synaptic element of a node (like Axonl boutons or
+   dendritic spines) used for structural plasticity.
+   The synaptic elements represent connection points between two neurons. They
+   grow according to a homeostatic growth rule. The dynamics of the
+   number of synaptic elements is driven by the average electrical activity of
+   the neuron (indirectly measured through the Calcium concentration of the
+   node). The probability of two neurons creating a new synapse between them,
+   depends on the number of available synaptic elements of each neuron.
+
+  Parameters:
+   z                double  - Current number of synaptic elements
+   continuous       boolean - Defines if the number of synaptic elements should
+                              be treated as a continuous double number or as an
+                              integer value. Default is false.
+   growth_rate      double  - The maximum amount by which the synaptic elements will
+                              change between time steps. In elements/update.
+   tau_vacant       double  - Rate at which vacant synaptic elements will decay.
+                              Typical is 0.1 which represents a
+                              loss of 10% of the vacant synaptic elements each time
+                              the structural_plasticity_update_interval is
+                              reached by the simulation time.
+   growth_curve     GrowthCurve* - Rule which defines the dynamics of this synaptic element.
```

Comment:
The type of `growth_curve` is string, valid strings are keys in `growthcurvedict`.